src/source-control/GitSourceControl.ts

```diff
@@ -230,10 +234,19 @@ export class GitSourceControl implements SourceControl {
       // 2. Update Target Commit to point to the cherry picked commit
       // 3. Update rebaseRootCommit to point to child of itself
 
-      const hashToBeRebased = queue.pop()!;
+      const hashOfCommitToBeRebased = queue.pop()!;
+      // `hashOfCommitToBeRebased` should always be present in the rebaseTargetHashMap, adding nullthrows here, so that if null,
+      // rebase cannot happen
+      targetCommitHash = nullthrows(
+        rebaseTargetHashMap.get(hashOfCommitToBeRebased),
+        `Target commit for ${hashOfCommitToBeRebased.slice(
+          0,
+          6,
+        )} could not be found `,
```

Comment:
Nit: should we dump the slicing since we don't truncate hashes in other errors?

```suggestion
        `Target commit for ${hashOfCommitToBeRebased} could not be found`,
```

Reply:
Yeah, I think I just did that to make it readable. I actually think we should truncate all the error logs to display truncated commit hashes.